packages/dolmen/dolmen.0.7/opam

```diff
@@ -11,7 +11,7 @@ depends: [
   "menhir" {>= "20211230" }
   "dune" { >= "2.7" }
   "fmt" { >= "0.8.7" }
-  "seq"
+  "seq" { = "base" }
```

Comment:
This feels wrong. Seq should just make an alias of Stdlib.Seq for ocaml > 4.07. Why it fails from 4.12 onwards?
@kit-ty-kate do you understand why downgrading to `seq.0.2.2` gives rise to
```

#=== ERROR while compiling dolmen_type.0.5 ====================================#
# context              2.1.2 | linux/x86_64 | ocaml-base-compiler.4.14.0 | file:///home/opam/opam-repository
# path                 ~/.opam/4.14/.opam-switch/build/dolmen_type.0.5
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p dolmen_type -j 31
# exit-code            1
# env-file             ~/.opam/log/dolmen_type-24-0fa409.env
# output-file          ~/.opam/log/dolmen_type-24-0fa409.out
### output ###
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlc.opt -w -40 -g -bin-annot -I src/typecheck/.dolmen_type.objs/byte -I /home/opam/.opam/4.14/lib/dolmen -I /home/opam/.opam/4.14/lib/dolmen/intf -I /home/opam/.opam/4.14/lib/dolmen/smtlib2 -I /home/opam/.opam/4.14/lib/dolmen/std -I /home/opam/.opam/4.14/lib/dolmen/tptp -I /home/opam/.opam/4.14/lib/spelll -I /home/opam/.opam/4.14/lib/uutf -intf-suffix .ml -no-alias-deps -open Dolmen_type -o src/typecheck/.dolmen_type.objs/byte/dolmen_type__Misc.cmo -c -impl src/typecheck/misc.ml)
# File "src/typecheck/misc.ml", line 241, characters 23-48:
# 241 |     match seq_to_list_ (I.retrieve ~limit:0 t s) with
#                              ^^^^^^^^^^^^^^^^^^^^^^^^^
# Error: This expression has type 'a Seq/1.t
#        but an expression was expected of type
#          'b Stdlib.Seq.t = unit -> 'b Stdlib.Seq.node
#        Seq/1.t is abstract because no corresponding cmi file was found in path.
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlopt.opt -w -40 -g -I src/typecheck/.dolmen_type.objs/byte -I src/typecheck/.dolmen_type.objs/native -I /home/opam/.opam/4.14/lib/dolmen -I /home/opam/.opam/4.14/lib/dolmen/intf -I /home/opam/.opam/4.14/lib/dolmen/smtlib2 -I /home/opam/.opam/4.14/lib/dolmen/std -I /home/opam/.opam/4.14/lib/dolmen/tptp -I /home/opam/.opam/4.14/lib/spelll -I /home/opam/.opam/4.14/lib/uutf -intf-suffix .ml -no-alias-deps -open Dolmen_type -o src/typecheck/.dolmen_type.objs/native/dolmen_type__Misc.cmx -c -impl src/typecheck/misc.ml)
# File "src/typecheck/misc.ml", line 241, characters 23-48:
# 241 |     match seq_to_list_ (I.retrieve ~limit:0 t s) with
#                              ^^^^^^^^^^^^^^^^^^^^^^^^^
# Error: This expression has type 'a Seq/1.t
#        but an expression was expected of type
#          'b Stdlib.Seq.t = unit -> 'b Stdlib.Seq.node
#        Seq/1.t is abstract because no corresponding cmi file was found in path.
```

Reply:
It may be due to the use of `(implicit_transitive_deps false)`, if I looked at the correct tarball

Reply:
In that case forcibly require the compiler seq is probably the best approach. I'll wait for @kit-ty-kate to correct me if I am wrong or merge